Issue 136 vmec and desc wrappers

pbonofig opened the pull request:
I added three routines within the B_STS class to translate VMEC and DESC free boundary output files and VMEC_EXTENDER output files to the needed B_STS input. Mainly, this is translating from VMEC/DESC coordinates to (R,phi,Z). The routines have been named 'vmec_field,' 'desc_field,' and 'extender_field.' These routines use the subroutines 'costransform' and 'sintransform,' which were also added. 

I note that the first two routines are only for free-boundary equilibria and are not valid for rho>1. It is up to the user to not simulate beyond this boundary, but the user should be aware of what equilibrium they are supplying in the first place. The extender version does allow for a B-field in the vacuum region but rho is ill-defined there. 

I have tested it on my equilibria, and they work. Unfortunately, I cannot share my equilibria. I hope other users have VMEC and DESC equilibria to help test. 

Lastly, the routines require the netCDF4 and desc-opt python modules, so I updated the environment .yaml files. I found that I had to place these modules within the 'pip' section for them to be found.

miekkasarki commented:
> I'm bit undecided on whether to include these new dependencies for all environments. For environment-dev.yaml for sure, but having already a heavy dependency like pyvista has taught me to be careful when introducing more. Do you know if netCDF and desc-opt are heavy? It might be enough that you make them optional in the Python modules like I've done here for example:

Related to this point, we might consider having a subpackage e.g. `interfaces` that would have all import/export routines and associated dependencies. This way one could pick and choose what to install based on their needs, e.g.:

```
pip install a5py
cd a5py
pip install interfaces[vmec]
```

But this won't concern this PR. I just wanted to wrote this down somewhere but not make an issue yet.

pbonofig commented:
Done.

I moved the routines to importdata.py. I also added two try statements at the the top for the desc and netcdf4 modules and importerror raises in the function calls themselves. 

I agree that the handling of the additional libraries is the biggest concern here, which is why I explicitly stated it. The netcdf4 module does not seem too heavy, but desc-opt does as it has a bunch of plotting functions and more. 

As I said, I followed your first suggestion and implemented try and except statements. Feel free to edit and remove the modules from the environment.yaml files (except the dev version). 

Let me know if there is anything else that I should change.

miekkasarki commented:
Thanks! Ok if desc-opt is potentially heavy, I'll remove it from the user environments just in case. However, I'll do it in a separate commit and merge this one right away.
